server/main.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"net/url"
+	"os"
+	"strconv"
+
+	"github.com/vertexcover-io/locatr"
+)
+
+var clientAndLocatrs = make(map[string]locatr.LocatrInterface)
+
+func createLocatrOptions(message incomingMessage) locatr.BaseLocatrOptions {
+	opts := locatr.BaseLocatrOptions{}
+	llmConfig := message.Settings.LlmSettings
+
+	if llmConfig.ReRankerApiKey != "" {
+		opts.ReRankClient = locatr.NewCohereClient(llmConfig.ReRankerApiKey)
+	}
+
+	opts.LogConfig = locatr.LogConfig{Level: locatr.Debug}
+
+	opts.CachePath = message.Settings.CachePath
+	opts.UseCache = message.Settings.UseCache
+
+	opts.ResultsFilePath = message.Settings.ResultsFilePath
+
+	llmClient, _ := locatr.NewLlmClient(
+		locatr.LlmProvider(llmConfig.LlmProvider),
+		llmConfig.ModelName,
+		llmConfig.LlmApiKey,
+	)
+	opts.LlmClient = llmClient
+
+	return opts
+}
+
+func handleLocatrRequest(message incomingMessage) (string, error) {
+	baseLocatr, ok := clientAndLocatrs[message.ClientId]
+	if !ok {
+		return "", fmt.Errorf("Client not instianciated of id: %s", message.ClientId)
+	}
+	locatr, err := baseLocatr.GetLocatrStr(message.UserRequest)
+	if err != nil {
+		return "", fmt.Errorf("Failed to retrieve locatr: %w", err)
+	}
+	return locatr, nil
+
+}
+
+func handleInitialHandshake(message incomingMessage) error {
+	baseLocatrOpts := createLocatrOptions(message)
+	switch message.Settings.PluginType {
+	case "cdp":
+		parsedUrl, _ := url.Parse(message.Settings.CdpURl)
+		port, _ := strconv.Atoi(parsedUrl.Port())
+		connectionOpts := locatr.CdpConnectionOptions{
+			Port:     port,
+			HostName: parsedUrl.Hostname(),
+		}
+		connection, err := locatr.CreateCdpConnection(connectionOpts)
+		if err != nil {
+			return fmt.Errorf("Error while creating cdp connection: %w", err)
+		}
+		cdpLocatr, err := locatr.NewCdpLocatr(connection, baseLocatrOpts)
+		if err != nil {
+			return fmt.Errorf("Error while creating cdp locatr: %w", err)
+		}
+		clientAndLocatrs[message.ClientId] = cdpLocatr
+	case "selenium":
+		settings := message.Settings
+		seleniumLocatr, err := locatr.NewRemoteConnSeleniumLocatr(settings.SeleniumUrl, settings.SeleniumSessionId, baseLocatrOpts)
+		if err != nil {
+			return fmt.Errorf("Error while creating selenium locatr: %w", err)
+		}
+		clientAndLocatrs[message.ClientId] = seleniumLocatr
+	}
+	return nil
+}
+
+func acceptConnection(fd net.Conn) {
+	defer fd.Close()
```

Comment:
it shouldn't be the responsibility of the function to close the objects that were passed in, as they don't know the entire lifecycle of the object. It should be the responsibility of the caller.